kubespawner/objects.py

```diff
@@ -511,46 +512,43 @@ def make_ingress(
         host, path = routespec.split('/', 1)
 
     target_parts = urlparse(target)
-
-    target_ip = target_parts.hostname
     target_port = target_parts.port
 
-    target_is_ip = re.match('^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$', target_ip) is not None
+    try:
+        # Try to parse as an IP address
+        target_ip = ipaddress.ip_address(target_parts.hostname)
```

Comment:
This is catching a possible ValueError in several places and treating them as invalid ip. Maybe just catch the parsing of the address?

Or am I missing which things might raise ValueError and why?

```suggestion
    try:
        # Try to parse as an IP address
        target_ip = ipaddress.ip_address(target_parts.hostname)
    except ValueError:
        target_is_ip = False
    else:
        target_is_ip = True
```

Reply:
Hi @minrk 

I addressed this in 2c3d9f5